frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -543,11 +553,12 @@ export const StyledBashCode = styled.code({
  * retries forever until one of the URIs responds with 'ok'.
  * Returns a promise with the index of the URI that worked.
  */
-export function doHealthPing(
-  uriList: string[],
+export function doInitPings(
```

Comment:
Oh my goodness this function is a nightmare (though not because of anything you've done here)